rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -217,14 +211,18 @@ object Reduce {
           case _ => s.unit
         }
 
-      List(
+      val jobs = List(
         EvalJob(par.sends, mkTermHandler[Send](evalExplicit)),
         EvalJob(par.receives, mkTermHandler[Receive](evalExplicit)),
         EvalJob(par.news, mkTermHandler[New](evalExplicit)),
         EvalJob(par.matches, mkTermHandler[Match](evalExplicit)),
         EvalJob(par.bundles, mkTermHandler[Bundle](evalExplicit)),
         EvalJob(filteredExprs, exprHandler)
-      ).zipWithIndex
+      )
+
+      val starts = jobs.map(_.size).scanLeft(0)(_ + _).toVector
```

Comment:
You've just convinced me the EvalJob has merit, due to it's size property :joy:

Reply:
Good, because I was preparing the response for your previous comment :)